Library/Homebrew/install.rb

```diff
@@ -32,12 +32,16 @@ def perform_build_from_source_checks(all_fatal: false)
     def check_prefix
       if (Hardware::CPU.intel? || Hardware::CPU.in_rosetta2?) &&
          HOMEBREW_PREFIX.to_s == HOMEBREW_MACOS_ARM_DEFAULT_PREFIX
-        configuration = if Hardware::CPU.in_rosetta2?
-          "under Rosetta 2"
+        if Hardware::CPU.in_rosetta2?
+          configuration = "under Rosetta 2"
+          odie <<~EOS
+            Cannot install in Homebrew #{configuration} in ARM default prefix (#{HOMEBREW_PREFIX})!
+            Try `arch -arm64 brew install ...`
```

Comment:
Can this be reworded away from "try" (which has a slight implication that it may not work). Does this sometimes/always work? If not/when not: why not?

Reply:
I can say "Use". I used try because I don't know if people are calling brew from other applications where they don't have control over the invocation. But I guess then this message works just as well for developers of those applications.